### ISSUE TITLE
SNL March 2016 bad data fails to load episodes

### ISSUE THREAD

**Issue by rsimac** [learningit/plugin.video.wnbc#2]
Pretty sure this is caused by bad input data, but right now (Apr 2016), the SNL Episodes are not listing, crashing as logged below.

I nailed and fixed (provisionally) the problem down to the following in:
C:\Users\user\AppData\Roaming\Kodi\addons\plugin.video.wnbc\resources\lib\scraper.py

I added the following skip of bad/missing input at line 66, the listings work ok after this, missing the bad data episode ("Ryan Gosling" I think)...
```
if not b['videoURL']:
            xbmc.log("videoURL empty for b: ")
            xbmc.log(str(b))
            continue
```

The original crash was as follows:

`19:31:23 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/r_S1Navy7BNb?format=preview
19:31:24 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/4TMmU6MdwIxp?format=preview
19:31:24 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/Vk2eW2AcahOT?format=preview
19:31:24 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/OjBCBoXqWU4_?format=preview
19:31:25 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/5CDXh2IGwOfe?format=preview
19:31:25 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/ok32JJl7e_YQ?format=preview
19:31:26 T:3200  NOTICE: html: http://link.theplatform.com/s/HNK2IC/_0XLZvgbbfuB?format=preview
19:31:27 T:3200  NOTICE: html: ?format=preview
19:31:27 T:3200   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: No JSON object could be decoded
                                            Traceback (most recent call last):
                                              File "C:\Users\user\AppData\Roaming\Kodi\addons\plugin.video.wnbc\default.py", line 12, in <module>
                                                ma.processAddonEvent()
                                              File "C:\Users\user\AppData\Roaming\Kodi\addons\script.module.t1mlib\lib\t1mlib.py", line 213, in processAddonEvent
                                                elif mode=='GE':  self.procDir(self.getAddonEpisodes,p('url'), 'episodes', 'episode_view')
                                              File "C:\Users\user\AppData\Roaming\Kodi\addons\script.module.t1mlib\lib\t1mlib.py", line 156, in procDir
                                                ilist = dirFunc(url, ilist)
                                              File "C:\Users\user\AppData\Roaming\Kodi\addons\plugin.video.wnbc\resources\lib\scraper.py", line 77, in getAddonEpisodes
                                                c = json.loads(html)
                                              File "E:\kodi\system\python\Lib\json\__init__.py", line 338, in loads
                                                return _default_decoder.decode(s)
                                              File "E:\kodi\system\python\Lib\json\decoder.py", line 366, in decode
                                                obj, end = self.raw_decode(s, idx=_w(s, 0).end())
                                              File "E:\kodi\system\python\Lib\json\decoder.py", line 384, in raw_decode
                                                raise ValueError("No JSON object could be decoded")
                                            ValueError: No JSON object could be decoded
                                            -->End of Python script error report<--
`

Hope this all makes sense to you, and this is the right way to report problems. I havent had time to browse the forums and such...

Thanks much for great plugin and let me know if you need more data...

**Comment by learningit** [comment]
Please report this in the Kodi forum. Put the entire debug log on pastebin.com or xbmclogs.com. Log snippets are useless and just clutter github and thr forum.

**Comment by rsimac** [comment]
Sorry man dont have time for forums, you can use the fix as is, or ignore
it.
The error will probably go away when nbc data gets fixed so it is not a big
issue.
Thanks for plugin...

On Mon, Apr 18, 2016 at 7:54 PM, learningit <notifications@github.com>
wrote:

> Closed #2 <https://github.com/learningit/plugin.video.wnbc/issues/2>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/learningit/plugin.video.wnbc/issues/2#event-632142441>
>



-- 
https://www.linkedin.com/in/rsimac
(202) 670-2766
